src/domain/ports/inbound/items/ports.py

```diff
@@ -0,0 +1,26 @@
+from typing import Protocol
+
+from src.domain.ports.inbound.items.dtos import (
+    AddItemInputDTO,
+    AddItemOutputDTO,
+    RemoveItemInputDTO,
+    RemoveItemOutputDTO,
+    SetInventoryQuantityInputDTO,
+    SetInventoryQuantityOutputDTO,
+)
+
+
+class AddItemPort(Protocol):
+    def execute(self, input_dto: AddItemInputDTO) -> AddItemOutputDTO: ...
+
+
+class RemoveItemPort(Protocol):
+    def execute(
+        self, input_dto: RemoveItemInputDTO
+    ) -> RemoveItemOutputDTO: ...
+
+
+class SetInventoryQuantityPort(Protocol):
+    def execute(
+        self, input_dto: SetInventoryQuantityInputDTO
+    ) -> SetInventoryQuantityOutputDTO: ...
```

Comment:
Cara, da pra fazer isso de um jeito um pouco mais idiota mas que economiza código talvez (caso surjam mais e mais casos de uso):
```suggestion
from typing import Generic, Protocol, TypeVar

InputDto = TypeVar("InputDto", contravariant=True)
OutputDto = TypeVar("OutputDto", covariant=True)


class UseCaseInterface(Protocol, Generic[InputDto, OutputDto]):
    async def execute(self, input_dto: InputDto) -> OutputDto:
        pass
```


Dessa forma, nos adaptadores inbound você tipa apenas que é um caso de uso, que "come" X e cospe "Y". Não recomendo, mas é uma forma divertida que eu pensei aqui pq durmi 4h essa noite e to chapado de privação de sono.